### PR TITLE
Feature: Dynamic JSON input for Streamlit recipes

### LINKED ISSUE #1: Enable dynamic user input of recipes in Streamlit app (paste JSON instead of file upload)
## Feature
Allow users to paste or type any recipe JSON into a text area in the Streamlit interface, rather than requiring file upload.

## Motivation
- Current interface is limited: user must upload a file, or choose only from hardcoded samples.
- More dynamic and demo-friendly if user can quickly paste in any recipe (including examples not in sample_recipes.json).
- Makes it easier to test edge cases or arbitrary recipes.

## Acceptance Criteria
- Users can paste or type valid JSON containing one or more recipes.
- The dropdown updates dynamically with available recipe names from user input.
- App still works with default sample_recipes.json if input is empty.
- If invalid JSON is entered, show an error message.

Image

### LINKED ISSUE #2: Add GitHub Actions workflow status badge to README.md
## Feature
Display the status of our CI/CD (linting) workflow as a badge at the top of README.md.

## Motivation
- Readers, maintainers, and reviewers can instantly see whether the latest code passes lint and CI checks.
- Best practice for all collaborative/public projects.

## Acceptance Criteria
- A badge appears in README.md reflecting the current status of our GitHub Actions workflow.
- Badge links to the Actions page for easy access.

### PR DESCRIPTION
**Pull request by teohsinyee**
## What’s Changed

- Added ability for users to paste or upload any recipe JSON directly in the Streamlit app
- Dropdown and analysis now dynamically update for arbitrary recipe input
- Improved error handling for invalid JSON
- Added new Korean recipes (Bibimbap, Kimchi Jjigae) as examples

## Why

- Makes the demo much more interactive and realistic
- Allows easy testing of new, edge-case, or user-supplied recipes

## Linked Issues

- Closes #1
- Closes #2

**Before:**
image

**After:**
image

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -9,27 +9,61 @@
...
-    recipes = json.load(uploaded)
-else:
-    # Fallback: load example recipes
+# --- Input area ---
+st.header("Input Recipes")
+
+tab1, tab2 = st.tabs(["Paste JSON", "Upload File"])
+
+recipes = None
+input_error = None
```
Comment (teohsinyee) [inline app.py:25]: this is something wrong!!! pls change
